src/snowflake/snowpark/modin/plugin/_internal/type_utils.py

```diff
@@ -224,7 +224,7 @@ class TypeMapper:
     @classmethod
     def to_snowflake(
         cls, p: Union[np.dtype, ExtensionDtype, native_pd.Timestamp]
-    ) -> DataType:
+    ) -> Union[DataType, SnowparkPandasType]:
```

Comment:
```suggestion
    ) -> DataType:
```

Each subclass of `SnowparkPandasType` should be a subclass of `DataType`.

Reply:
I'll add it.